modules/join/src/main/java/org/apache/ignite/internal/join/Utils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.join;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.ClusterService;
+
+/**
+ * Various utility methods.
+ */
+class Utils {
```

Comment:
This seems to be a too vague name. I suggest specifying what kind of util it is, like `NodeResolveUtils` or `NodeUtils`, or even dropping the `Utils` and calling it `NodeResolving`.

Reply:
calling it `NodeResolving` will imply that all methods in this class must somehow relate to "resolving". I'm not particularly sure about the future of this class, because having utility classes is bad, and it also may obtain more methods in future PRs, not related to resolving.

Reply:
Well, right now its name says nothing about the code that resides in it. At least now it is about resolving nodes, so now `NodeResolving` (or something similar) is the most adequate name. In the future, if other methods are added, the class can be renamed accordingly. Or the methods should be added to another class.